### PR TITLE
IntelliJ inspections cleanup

### PR DESCRIPTION
**Pull request by suneet-s**
* Standard Charset object can be used
* Redundant Collection.addAll() call
* Statement with empty body
* Redundant Collection operation
* StringBuilder can be replaced with String
* Type parameter hides visible type

Reading through the warnings, none of these appear terribly bad. 

The type parameter warning is a good one to be an error since it's usually a mistake in parameterizing the types by the dev.
Empty statement bodies are also good errors to flag since it's possible an empty body
was unintentional.

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/query/groupby/having/GreaterThanHavingSpec.java [hunk 1]
```diff
@@ -121,12 +121,10 @@ public int hashCode()
...
   public String toString()
   {
-    final StringBuilder sb = new StringBuilder();
-    sb.append("GreaterThanHavingSpec");
-    sb.append("{aggregationName='").append(aggregationName).append('\'');
-    sb.append(", value=").append(value);
-    sb.append('}');
-    return sb.toString();
+    return "GreaterThanHavingSpec{" +
+           "{aggregationName='" + aggregationName + '\'' +
```
Comment (ccaominh) [inline processing/src/main/java/org/apache/druid/query/groupby/having/GreaterThanHavingSpec.java]: There's an extra `{` that got added here.

Comment (suneet-s) [reply processing/src/main/java/org/apache/druid/query/groupby/having/GreaterThanHavingSpec.java]: Done

File: processing/src/main/java/org/apache/druid/segment/writeout/OnHeapMemorySegmentWriteOutMediumFactory.java [hunk 2]
```diff
@@ -23,6 +23,7 @@
 
 import java.io.File;
 
+
```
Comment (ccaominh) [inline processing/src/main/java/org/apache/druid/segment/writeout/OnHeapMemorySegmentWriteOutMediumFactory.java:26]: Revert?

File: services/src/main/java/org/apache/druid/cli/CliHadoopIndexer.java [hunk 3]
```diff
@@ -84,8 +84,8 @@ public void run()
         extensionURLs.addAll(Arrays.asList(((URLClassLoader) extensionLoader).getURLs()));
       }
 
-      final List<URL> nonHadoopURLs = new ArrayList<>();
-      nonHadoopURLs.addAll(Arrays.asList(((URLClassLoader) CliHadoopIndexer.class.getClassLoader()).getURLs()));
+      final List<URL> nonHadoopURLs = new ArrayList<>(Arrays.asList(((URLClassLoader) CliHadoopIndexer.class.getClassLoader())
```
Comment (ccaominh) [inline services/src/main/java/org/apache/druid/cli/CliHadoopIndexer.java:87]: The `new ArrrayList<>()` seems unnecessary?

File: processing/src/main/java/org/apache/druid/segment/writeout/OnHeapMemorySegmentWriteOutMediumFactory.java (hunk 2 above)
Comment (suneet-s) [reply processing/src/main/java/org/apache/druid/segment/writeout/OnHeapMemorySegmentWriteOutMediumFactory.java:26]: removed

File: services/src/main/java/org/apache/druid/cli/CliHadoopIndexer.java (hunk 3 above)
Comment (suneet-s) [reply services/src/main/java/org/apache/druid/cli/CliHadoopIndexer.java]: removed

**Pull request merged**